pilot/pkg/security/authn/utils/utils.go

```diff
@@ -69,11 +71,29 @@ func BuildInboundTLS(mTLSMode model.MutualTLSMode, node *model.Proxy,
 
 	// Set Minimum TLS version to match the default client version and allowed strong cipher suites for sidecars.
 	ctx.CommonTlsContext.TlsParams = &tls.TlsParameters{
-		TlsMinimumProtocolVersion: tls.TlsParameters_TLSv1_2,
-		CipherSuites:              SupportedCiphers,
+		CipherSuites: SupportedCiphers,
 	}
+	// Configure TLS version based on meshconfig TLS API.
+	// The min TLS version is initialized to be at least TLS 1.2.
+	minTLSVersion := meshconfig.MeshConfig_TLSConfig_TLSV1_2
+	if mc != nil && mc.GetMeshMTLS() != nil {
+		minTLSVersion = mc.GetMeshMTLS().MinProtocolVersion
+	}
+	configureTLSVersion(minTLSVersion, ctx.CommonTlsContext.TlsParams)
 
 	authn_model.ApplyToCommonTLSContext(ctx.CommonTlsContext, node, []string{}, /*subjectAltNames*/
 		trustDomainAliases, ctx.RequireClientCertificate.Value)
 	return ctx
 }
+
+func configureTLSVersion(minTLSVer meshconfig.MeshConfig_TLSConfig_TLSProtocol, tlsParam *tls.TlsParameters) {
```

Comment:
The code could be cleaner if you just pass the `mc` and return the min/max TLS version directly from the function, like `getTLSMinMaxVersion(mc *meshconfig.MeshConfig) (minVersion TlsParameters_TlsProtocol, maxVersion TlsParameters_TlsProtocol)`

Reply:
The original function is refactored into getTLSVersions(mcMinTLSVer meshconfig.MeshConfig_TLSConfig_TLSProtocol) (tls.TlsParameters_TlsProtocol, tls.TlsParameters_TlsProtocol).